src/tokens.js

```diff
@@ -702,15 +718,48 @@ const tokens = {
    * @param {Object} data Array of objects each one containing the input with the authority being destroyed ({'tx_id', 'index', 'address', 'token'})
    * @param {string} pin Pin to generate new addresses, if necessary
    *
-   * @return {Promise} Promise that resolves when transaction destroying the authority is completed
+   * @return {Object} In case of success, an object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   * In case of error, an object with {success: false, message}
    *
    * @memberof Tokens
    * @inner
    */
   destroyAuthority(data, pin) {
     // Create new data without any output
     let newTxData = {'inputs': data, 'outputs': [], 'tokens': []};
-    return transaction.sendTransaction(newTxData, pin);
+    return this.handleSendTransaction(newTxData, pin);
+  },
+
+  /**
+   * Get transaction data, prepare it, create a SendTransaction object and a promise that succeeds and fails depending on the object events
+   *
+   * @param {Object} data Array of objects each one containing the input with the authority being destroyed ({'tx_id', 'index', 'address', 'token'})
+   * @param {string} pin Pin to generate new addresses, if necessary
+   * @param {Object} options {
+   *   {number} minimumTimestamp Tx minimum timestamp (default = 0)
+   *   {boolean} createAnotherMint If should create another mint output after spending this one
+   *   {boolean} createMelt If should create a melt output (useful when creating a new token)
+   * }
+   *
+   * @return {Object} In case of success, an object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   * In case of error, an object with {success: false, message}
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  handleSendTransaction(data, pin, options) {
+    let preparedData = null;
+    try {
+      preparedData = transaction.prepareData(data, pin, options);
+    } catch (e) {
+      const message = helpers.handlePrepareDataError(e);
```

Comment:
Why do we need `const message = helpers.handlePrepareDataError(e); `? We don't use it.

Reply:
My bad, the returned message was supposed to be `message` and not `e.message`. Fixed it.